PDB file not copied to targetPath alongside executable on Windows

### System information
- **dub version**: `DUB version 1.20.1, built on Apr 17 2020`
- **OS Platform and distribution**: `Microsoft Windows [Version 10.0.18363.535]` -- (1909), 64-bit
- **compiler version**: `dmd-2.091.1`

### Bug Description
When building an executable the `.exe` is copied to the `targetPath` however the `.pdb` file with the debugging information is not.

### How to reproduce?
```
> type source\app.d
void main() {}

> type dub.sdl
name "foo"
targetName "foo-bin"
targetType "executable"
targetPath "bin\\64"

> dub build
...

> dir /b bin\64
foo-bin.exe
```

### Expected Behavior
```
> dir /b bin\64
foo-bin.exe
foo-bin.pdb
```

### Use Case
On Windows it's common for in-house software to ship the `.pdb` files along with the `.exe` to make live debugging easier. Alternatively the `.pdb` files are published to a [Symbol Server](https://en.wikipedia.org/wiki/Microsoft_Symbol_Server) where they can be downloaded on-the-fly when debugging. In both scenarios we need easy access to the `.pdb` file rather than spelunking the `.dub\build` folder.

One wrinkle I've run into with the `lflags "/PDB:bin\\64\\app.pdb"` approach is that the target folder needs to exist _before_ you start building otherwise you end up failing to link with:
```
Linking...
LINK : fatal error LNK1201: error writing to program database 'C:\...\bin\64\app.pdb'; check for insufficient disk space, invalid path, or insufficient privilege
Error: linker exited with status 1201
```
At the moment `dub` appears to create the `targetPath` folder _after_ linking.

--- a/changelog/pdb.dd
+++ b/changelog/pdb.dd
@@ -1,0 +1,5 @@
+Windows: Copy PDB files to `targetPath`, alongside executable/DLL
+
+If the default PDB file is generated when linking an executable or DLL
+(e.g., no `/PDB:my\path.pdb` lflag), it is now copied to the target
+directory too.

--- a/source/dub/generators/build.d
+++ b/source/dub/generators/build.d
@@ -371,12 +371,30 @@ class BuildGenerator : ProjectGenerator {
 
 	private void copyTargetFile(NativePath build_path, BuildSettings buildsettings, GeneratorSettings settings)
 	{
-		auto filename = settings.compiler.getTargetFileName(buildsettings, settings.platform);
-		auto src = build_path ~ filename;
-		logDiagnostic("Copying target from %s to %s", src.toNativeString(), buildsettings.targetPath);
 		if (!existsFile(NativePath(buildsettings.targetPath)))
 			mkdirRecurse(buildsettings.targetPath);
-		hardLinkFile(src, NativePath(buildsettings.targetPath) ~ filename, true);
+
+		string[] filenames = [
+			settings.compiler.getTargetFileName(buildsettings, settings.platform)
+		];
+
+		// Windows: add .pdb if found
+		const tt = buildsettings.targetType;
+		if ((tt == TargetType.executable || tt == TargetType.dynamicLibrary) &&
+		    settings.platform.platform.canFind("windows"))
+		{
+			import std.path : setExtension;
+			const pdbFilename = filenames[0].setExtension(".pdb");
+			if (existsFile(build_path ~ pdbFilename))
+				filenames ~= pdbFilename;
+		}
+
+		foreach (filename; filenames)
+		{
+			auto src = build_path ~ filename;
+			logDiagnostic("Copying target from %s to %s", src.toNativeString(), buildsettings.targetPath);
+			hardLinkFile(src, NativePath(buildsettings.targetPath) ~ filename, true);
+		}
 	}
 
 	private bool isUpToDate(NativePath target_path, BuildSettings buildsettings, GeneratorSettings settings, in Package main_pack, in Package[] packages, in NativePath[] additional_dep_files)

